### ISSUE TITLE
Running the test suite deletes .gitkeep files in the infrastructure's file_system

### ISSUE THREAD

**Issue by Thomsch** [TellinaTool/tellina_user_experiment#18]
Running the test suite from the root using `make -C client_side/.infrastructure test` causes all the `.gitkeep` files in ` client_side/.infrastructure/file_system/` to be deleted.

This is problematic because executing the tests should not have the side effect of changing the development environment. The second reason is that the current setup relies on this side effect to avoid having diff conflicts during the experiment. Indeed, no directory during the experiment is supposed to have a .gitkeep inside it, otherwise, the task will not be validated.

**Comment by anhnamtran** [comment]
Thanks for catching this, any suggestion on how to fix this?

**Comment by mernst** [comment]
This is the test that deletes the files:
```
@test "next_task ends experiment" {
  local TASKS_SIZE=10
  local task_num=9

  run next_task

  assert_success
}
```

It appears in `client_side/.infrastructure/test/infrastructure_tests/task_handling_tests.bats`.

**Comment by Thomsch** [comment]
In addition to @mernst's suggestion, I think the two likely candidates are the tests in `reset_fs_tests.bats` and test `next_task resets the file system directory` in `task_handling_tests.bats`.

`reset_fs_tests.bats` tests' make calls to `reset_fs()`, which copies all files from a template structure **excluding '.*' files**:
```
reset_fs() {
  rsync --exclude=".*" --omit-dir-times --recursive --quiet --delete \
    --times \
    "${FS_SYNC_DIR}/" "${FS_DIR}"
}
```

The test `next_task resets the file system directory` has the line `find ${FS_DIR} -type f -delete`, which deletes all the files in the filesystem.

**Comment by anhnamtran** [comment]
Hmmm. This might be an oversight on my part, then. I think `reset_fs` should do the right thing, since it excludes `.*` files, so I think it might be `find ${FS_DIR} -type f -delete` is the culprit in removing the `.gitkeep`.

**Comment by mernst** [comment]
> In addition to @mernst's suggestion, I think the two likely candidates are

I ran the tests with and without that test commented out.  When that test is commented out, the files are not deleted.

**Comment by Thomsch** [comment]
I didn't realize you already tested it! Awesome. Thank you!

**Comment by mernst** [comment]
Sorry that was not clear.  It would have been clearer for me to explain my justification.

I would have said something like "maybe it's this test" or "I suspect it's this test" if I was just speculating.

**Comment by Thomsch** [comment]
It makes sense, especially when rereading it. I'll know for next time. Thank you!

**Comment by Thomsch** [comment]
Fixed the bug in commit 7d4176cade048ec0e2c79da70ca6998c180bc189.

The makefile now runs the tests on the distribution experiment itself and another make step remove the .gitkeep files since we don't want them in the experiment package.
